### PR TITLE
Logistics APIs

### PR DESCRIPTION
**Pull request by asim**

### CODE REVIEW

File: eta/proto/eta.proto [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+message Point {
+	double lat = 1;
+	double long = 2;
+}
+
+message Route {
+	Point start = 1;
+	Point finish = 2;
+	repeated Point waypoints = 3;
+	int64 start_at = 1;
```
Comment (ben-toogood) [inline eta/proto/eta.proto:18]: We should use the google timestamp here. I'd also add a vehicle type enum, e.g. is the route being done in a car or lorry (different speeds).

File: zoning/proto/zoning.proto [hunk 2]
```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
+
+service Zoning {
+	Create(Zone) returns (Empty) {};
+	Delete(Zone) returns (Empty) {};
+	Search(Query) returns (Result) {};
+	Subscribe(Point) returns (stream Event) {};
```
Comment (ben-toogood) [inline zoning/proto/zoning.proto:7]: I think given this is done over HTTP JSON, we should avoid streams. Webhooks could be an alternative.

File: routes/proto/routes.proto [hunk 3]
```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
+
+service Routes {
+	Guide(Query) returns (Route) {};
+	Record(Journey) returns (Empty) {};
```
Comment (ben-toogood) [inline routes/proto/routes.proto:5]: I feel like Record/Retrieve should be a part of the locations service, it's almost a duplication.

File: eta/proto/eta.proto [hunk 4]
```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
```
Comment (crufter) [inline eta/proto/eta.proto:3]: What does this do? I don't see it used anywhere, or just by importing it changes something?

File: points/README.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# Points Service
```
Comment (ben-toogood) [inline points/README.md:1]: This feels like it should be part of the location service

File: geocoding/proto/geocoding.proto [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+service Geocoding {
+	Geocode(Address) returns (Result) {};
+	Reverse(Point) returns (Result) {};
+}
+
+
+message Address {
+	string street = 1;
+	string postcode = 2;
```
Comment (ben-toogood) [inline geocoding/proto/geocoding.proto:11]: We'd need more info here, we could get the types from the Google API.

File: eta/proto/eta.proto [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+message Route {
+	Point start = 1;
+	Point finish = 2;
+	repeated Point waypoints = 3;
+	int64 start_at = 1;
+}
+
+message Result {
+	int64 arrival_time = 1;
+	int64 departure_time = 2;
```
Comment (ben-toogood) [inline eta/proto/eta.proto:23]: We'd also want to know the arrival & departure time for each waypoint.

File: eta/proto/eta.proto (hunk 1 above)
Comment (crufter) [reply eta/proto/eta.proto:18]: s/vehicle/mode of transport?
Unsure if there is a standard term for this, probably there is.

File: routes/proto/routes.proto (hunk 3 above)
Comment (crufter) [reply routes/proto/routes.proto:5]: I lean towards agreeing, Ben.

File: zoning/proto/zoning.proto (hunk 2 above)
Comment (crufter) [reply zoning/proto/zoning.proto:7]: We use streaming endpoints for logs on the dashboard and they work okayish.

File: points/proto/points.proto [hunk 8]
```diff
@@ -0,0 +1,16 @@
+syntax = "proto3";
+
+service Points {
+	Ingest(Batch) returns (Empty) {};
+	Record(Point) returns (Empty) {};
```
Comment (domwong) [inline points/proto/points.proto:5]: I'm guessing ingest is just the batch version of Record? Should either just make a single one that takes a batch (which you can pass a batch of length 1) or rename Ingest to `RecordBatch`

Comment (crufter) [reply points/proto/points.proto:5]: Hmm the list by default is a nice idea

File: zoning/proto/zoning.proto (hunk 2 above)
Comment (domwong) [reply zoning/proto/zoning.proto:7]: Is streaming something that most people are familiar with though? I feel like webhooks is pretty ubiquitous. Streaming? Not sure

**Pull request closed without merging**